modules/reitit-core/src/reitit/impl.cljc

```diff
@@ -9,6 +9,51 @@
      (:import (java.net URLEncoder URLDecoder)
               (java.util HashMap Map))))
 
+;;
+;; path-update
+;;
+
+(defn -match [path path-map]
+  (letfn [(match [x f] (if (fn? f) (f x) (= x f)))]
+    (reduce
+     (fn [_ [ps f]]
+       (let [match (loop [[p & pr] path, [pp & ppr] ps]
+                     (cond (and p pp (match p pp)) (recur pr ppr)
+                           (= nil p pp) true))]
```

Comment:
could this be simplified to something like `(every? identity (map match path ps))`?

Reply:
`map` uses the min length of the inputs, need to check that path is at least as long as ps. but, much better still!

```clojure
(and (>= (count path) (count ps))
     (every? identity (map match path ps)))
```